File: manual/code/generate_strategies/RA(method)/python/function_Schematics_model_import_data.py
from schematics.models import Model
from schematics.types import StringType, IntType

class Product(Model):
    name = StringType(required=True)
    price = IntType(required=True, min_value=0)
    category = StringType(default="General")

    def import_and_validate(self, data: str) -> str:
        try:
            data_dict = eval(data)
            self.import_data(data_dict, partial=False)
            self.validate()
            return str(self.to_native())
        except Exception as e:
            return f"error: {str(e)}"

File: code/generate_strategies/Direct/cpp/function_Schematics_model_import_data.cpp
#include <string>
#include <stdexcept>
#include <nlohmann/json.hpp>

using json = nlohmann::json;

class Product {
public:
    std::string import_and_validate(const std::string& data) {
        try {

            name = "";
            price = 0;
            category = "General";
            
  
            json data_dict = json::parse(data);
            
 
            if (data_dict.contains("name")) {
                name = data_dict["name"].get<std::string>();
            }
            
            if (data_dict.contains("price")) {
                price = data_dict["price"].get<int>();
            }
            
            if (data_dict.contains("category")) {
                category = data_dict["category"].get<std::string>();
            }
            

            if (name.empty()) {
                throw std::runtime_error("name: This field is required.");
            }
            if (price < 0) {
                throw std::runtime_error("price: Value must be greater than or equal to 0.");
            }
            
  
            std::string result = "{'name':'" + name + "','price':" + std::to_string(price) + 
                                ",'category':'" + category + "'}";
            
            return result;
        }
        catch (const json::parse_error& e) {
            return "error: Invalid JSON format";
        }
        catch (const std::exception& e) {
            return std::string("error: ") + e.what();
        }
    }

private:
    std::string name;
    int price;
    std::string category;
};